### PR TITLE
Power Divergence Test

### PR DESCRIPTION
**Pull request by bdeonovic**
Implements power divergence test, which is a family of tests for multinomial proportions that includes the chi-squared test. Also includes several popular multinomial confidence intervals (e.g. sison-glaz, bootstrap)

Since this is the first hypothesis test for a vector of parameters I wasn't quite sure how you guys would want the output summary presented. Let me know what you think!

### CODE REVIEW

File: src/power_divergence.jl [hunk 1]
```diff
@@ -0,0 +1,349 @@
+export PowerDivergenceTest, ChisqTest, MultinomialLRT
+
+boundproportion(x) = max(min(x,1),0)
+
+immutable MultinomialData
+  observed::Vector
```
Comment (johnmyleswhite) [inline src/power_divergence.jl]: These things should probably be parametric types. As is, they're black holes for the type inference system.

**Comment by johnmyleswhite** [comment]
Generally, this looks good to me.

**Comment by bdeonovic** [comment]
I tidied up the code, squashed commits together. I think this is ready to be pulled now.

**Comment by simonbyrne** [comment]
Great, thanks!

**Pull request merged**